types/src/v0/impls/state.rs

```diff
@@ -895,9 +1057,110 @@ mod test {
         eth_signature_key::{BuilderSignature, EthKeyPair},
         v0_1, v0_2,
         v0_3::{self, BidTx},
-        BlockSize, FeeAccountProof, FeeMerkleProof,
+        BlockSize, FeeAccountProof, FeeMerkleProof, Payload, Transaction,
     };
 
+    impl Transaction {
+        async fn into_mock_header(self) -> (Header, u32) {
+            let instance = NodeState::mock_v2();
+            let (payload, metadata) =
+                Payload::from_transactions([self], &instance.genesis_state, &instance)
+                    .await
+                    .unwrap();
+
+            let builder_commitment = payload.builder_commitment(&metadata);
+            let payload_bytes = payload.encode();
+
+            let payload_commitment = vid_commitment(&payload_bytes, GENESIS_VID_NUM_STORAGE_NODES);
+
+            let header =
+                Header::genesis(&instance, payload_commitment, builder_commitment, metadata);
+
+            let header = header.sign();
+
+            (header, payload.byte_len().0 as u32)
+        }
+    }
+    impl Header {
+        /// Build a new header from parent.
+        fn next(self) -> Self {
+            match self {
+                Header::V1(_) => panic!("You called `Header.next()` on unimplemented version (v1)"),
+                Header::V2(parent) => Header::V2(v0_2::Header {
+                    height: parent.height + 1,
+                    timestamp: OffsetDateTime::now_utc().unix_timestamp() as u64,
+                    ..parent.clone()
+                }),
+                Header::V3(_) => panic!("You called `Header.next()` on unimplemented version (v3)"),
+            }
+        }
+        /// Replaces builder signature w/ invalid one.
```

Comment:
```suggestion
        /// Replaces builder signature w/ a valid one from a random key pair
```
?

Reply:
The intent is convey that signature validation will fail because we are adding a signature that doesn't match the fee_info. It is therefore invalid (relative to what it will be validated against).

Reply:
Thinking again I see your point... The signature is structurally valid but will fail verification.